cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -615,6 +615,18 @@ data BuildResult = BuildResult {
        buildResultDocs    :: DocsResult,
        buildResultTests   :: TestsResult,
        buildResultLogFile :: Maybe FilePath,
+       -- | If the build was for a library, this field will be @Just@;
+       -- otherwise, it will be @Nothing@.  Note that for the legacy
+       -- install codepath, we employ a slight hack: only the
+       -- public library's 'InstalledPackageInfo' is stored here
+       -- (internal libraries are registered but not recorded
+       -- here.)  In the legacy code path, this 'InstalledPackageInfo'
+       -- is not used anyway, so it makes no difference.  This never
+       -- occurs for new-build, because a build result is either
+       -- for a per-component build (in which case there won't
+       -- be multiple libraries), or a package with no internal
+       -- libraries (internal libraries with Custom setups are NOT
+       -- supported).
```

Comment:
Note that currently this type is used only for new-build, the old code gets its own version of this type.